lib/task_bunny/job_error.ex

```diff
@@ -0,0 +1,97 @@
+defmodule TaskBunny.JobError do
+  @moduledoc """
+  A struct that holds an error information occured during the job processing.
+
+  ## Attributes
+
+  - job: the job module failed
+  - payload: the payload(arguments) for the job execution
+  - error_type: the type of the error. :exception, :return_value, :timeout or :exit
+  - exception: the inner exception (option)
+  - stacktrace: the stacktrace (only available for the exception)
+  - return_value: the return value from the job (only available for the return value error)
+  - reason: the reason information passed with EXIT signal (only available for exit error)
+  - raw_body: the raw body for the message
+  - meta: the meta data given by RabbitMQ
+  - failed_count: the number of failures for the job processing request
+  - queue: the name of the queue
+  - concurrency: the number of concurrent job processing of the worker
+  - pid: the process ID of the worker
+  - reject: sets true if the job is rejected for the failure (means it won't be retried again)
+
+  """
+
+  @type t :: %__MODULE__{
+    job: atom,
+    payload: any,
+    error_type: :exception | :return_value | :timeout | :exit | nil,
+    exception: struct | nil,
+    stacktrace: list(tuple) | nil,
+    return_value: any,
+    reason: any,
+    raw_body: String.t,
+    meta: map,
+    failed_count: integer,
+    queue: String.t,
+    concurrency: integer,
+    pid: pid,
+    reject: boolean
+  }
+
+  defstruct [
+    job: nil,
```

Comment:
The value of `:job` is defined as only `atom`, so it might be nicer to write `:job,` here in stead of `job: nil`. (Because that would be a default that is technically not allowed by the spec.)

If you go for `:job` then I would also add `@enforce_keys [:job]` so it also does not default to `nil`.

The same goes for `:pid`.